### PR TITLE
feat: Support `--deployment` and `--result` filters in history command

### LINKED ISSUE #7486: feat: Cloud CLI - enable filtering of schedule history by `deployment` and `result`
* ~~`meltano cloud history --name=<>`~~ Covered by https://github.com/meltano/meltano/pull/7467

* `meltano cloud history --deployment=<>`

* `meltano cloud history --result=[Success|Failed|Running]`

### PR DESCRIPTION
**Pull request by edgarrmondragon**
- Depends on https://github.com/meltano/infra/pull/777
- Closes #7486

### CODE REVIEW

File: src/cloud-cli/meltano/cloud/cli/history/command.py [hunk 1]
```diff
@@ -136,21 +157,32 @@ async def history(
...
+        deployment,
+        deployment_prefix,
+        deployment_contains,
+    )
 
     now = datetime.datetime.now(tz=utils.UTC)
     items = await HistoryClient.get_history_list(
         context.config,
         schedule_filter=schedule_filter,
+        environment_filter=deployment_filter,
```
Comment (tayloramurphy) [inline src/cloud-cli/meltano/cloud/cli/history/command.py:184]: This feels like a dangerous equality. At some point users might want to actually filter on the Meltano environment name across deployments. What would happen then? (I realize this is theoretical right now but a deployment is not 1 to 1 with an environment.

Comment (edgarrmondragon) [reply src/cloud-cli/meltano/cloud/cli/history/command.py:184]: Makes sense, I've started https://github.com/meltano/infra/pull/845 to actually be able to filter by deployment

**Review by tayloramurphy — APPROVED** [review]
Approving. I'm unable to test this locally even after installing from the branch though. What am I doing wrong?

**Comment by edgarrmondragon** [comment]
> Approving. I'm unable to test this locally even after installing from the branch though. What am I doing wrong?

@tayloramurphy can you try installing with

```
$ pipx install 'git+https://github.com/meltano/meltano.git@refactor/history-cli-jobs-api#subdirectory=src/cloud-cli'
```

**Comment by tayloramurphy** [comment]
@edgarrmondragon that's what I was using. If I uninstalled the cli and then reinstalled it worked. Approved!

**Pull request merged**